base/applications/rapps/gui.cpp

```diff
@@ -544,7 +544,7 @@ CMainWindow::OnCommand(WPARAM wParam, LPARAM lParam)
                 break;
 
             case ID_REFRESH:
-                UpdateApplicationsList(SelectedEnumType);
+                UpdateApplicationsList(SelectedEnumType, bReload);
```

Comment:
Should this do the full reload for DB items? [My PR](https://github.com/reactos/reactos/pull/5676) that this is a duplicate of only does the reload for Installed items, not the DB items (That is done by "Update Database"/Ctrl+F5?)

Reply:
No, this doesn't update the database. It's only update the installed items, works the same as before it's broken by the mentioned commit.

Reply:
Forcing the 2nd parameter of UpdateApplicationsList to be true instead of false is a change in behavior and will cause m_Db->UpdateAvailable() to be called when rapps is viewing non-installed applications. While DownloadApplicationsDB is unlikely to be called, it is going to reload all the .txt files from disk. If this is desired, I don't know.

Reply:
> it is going to reload all the .txt files from disk.

Yes, it will reload the app list from local db files when viewing the "Available for Installation" page. That is what it should do.

Reply:
Ok then. Perhaps you want to grab the selection restore part from my PR so I can delete mine and you can merge this?

Reply:
> Perhaps you want to grab the selection restore part from my PR

Since that's your purposed enhancement. I'll let you decide.
If there is any conflict with your PR, my changes are minimal so you can easily fix it.